### ISSUE TITLE
Timeline shows incorrect diff for elements in array

### ISSUE THREAD

**Issue by josh-** [medplum/medplum#4998]
The timeline in the medplum app shoes incorrect before/after values for values in arrays. For example, if I create this patient resource:

```json
{
  "resourceType": "Patient",
  "name": [
    {
      "given": [
        "Test"
      ]
    }
  ],
  "telecom": [
    {
      "value": "one"
    },
    {
      "value": "two"
    },
    {
      "value": "three"
    },
    {
      "value": "four"
    },
    {
      "value": "five"
    }
  ]
}
```

and then delete the "two" and "five" values in the medplum app, the timeline shows:

<img width="943" alt="image" src="https://github.com/user-attachments/assets/8ea705ff-e9e8-42c6-b39b-e67ac80f101e">

despite the underlying diff being this:

<img width="943" alt="image" src="https://github.com/user-attachments/assets/02f25629-91cc-46e6-bc21-09ed4e6098b8">

**demo video**

https://github.com/user-attachments/assets/04cd43e7-e25d-4450-be62-48bedd92542b

**Comment by codyebberson** [comment]
Thanks @josh-  - Yep, confirmed.  The timeline tries to be smart, looks like it failed in this case.

iirc, the component generates a JSONPatch between the before and after, and then tries to format the JSONPatch into a human friendly table.  I suspect what's going on here is that the JSONPatch is "stateful", in that it is a sequence of operations, and therefore when we try to translate "remove element 4", we would need to know that "remove element 2" had already happened.

I'm not sure that the JSONPatch strategy is actually best here.  Lots of complicated edge cases.  Might be better to do a more straightforward diff.

**Comment by josh-** [comment]
Yeah that makes sense given when we call `patchResource` we've also encountered json patch being stateful in this manner.

Atraightforward diff instead sounds good to me - thankyou 👍
